pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -924,12 +936,10 @@ impl RemoteTimelineClient {
             ))?
         });
 
-        let index_file_path = timeline_storage_path.join(Path::new(IndexPart::FILE_NAME));
-
         debug!("enqueuing index part deletion");
-        self.deletion_queue_client
-            .push_immediate([index_file_path].to_vec())
-            .await?;
+        if !indices.is_empty() {
+            self.deletion_queue_client.push_immediate(indices).await?;
+        }
```

Comment:
Should there still be something like preserving the latest version with the `deleted_at` to be deleted last, as in with a separate flush, so that we could always pick this up and more importantly never resurrect the timeline?

Or is this handled by something else now with the deletion queue...?

Reply:
Good point -- eventually it shouldn't matter, because we expect that the control plane will retry until the deletion works, but for the moment let's preserve the behavior of deleted_at acting like a tombstone.  I've updated this PR to only defer deletion of the most recent index, so that it should outlive all the others and preserve this behavior.